From 6f8136eb44240ef1d77d02d5119d689183831335 Mon Sep 17 00:00:00 2001
From: Dmitry Smirnov <divis1969@gmail.com>
Date: Mon, 14 Nov 2016 23:30:38 +0300
Subject: [PATCH 1/2] Remove CAP_SYS_NICE from surfaceflinger

This capability causes surfaceflinger to be treated as a secure process
(AT_SECURE) and linker ignores LD_PRELOAD and LD_SHIM_LIBS but we need these!

Change-Id: Ie9451a970bf43ec864b626e4d3e646bd17869c67
---
 libcutils/fs_config.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libcutils/fs_config.c b/libcutils/fs_config.c
index 68cfcbe..f192386 100644
--- a/libcutils/fs_config.c
+++ b/libcutils/fs_config.c
@@ -143,7 +143,8 @@ static const struct fs_path_config android_files[] = {
     { 00700, AID_SYSTEM,    AID_SHELL,     CAP_MASK_LONG(CAP_BLOCK_SUSPEND), "system/bin/inputflinger" },
 
     /* Support FIFO scheduling mode in SurfaceFlinger. */
-    { 00755, AID_SYSTEM,    AID_GRAPHICS,     CAP_MASK_LONG(CAP_SYS_NICE), "system/bin/surfaceflinger" },
+    /* Disabled for now as this caused LD_PRELOAD and LD_SHIM_LIBS to be sanitized in linker */
+    //{ 00755, AID_SYSTEM,    AID_GRAPHICS,     CAP_MASK_LONG(CAP_SYS_NICE), "system/bin/surfaceflinger" },
 
     { 00750, AID_ROOT,      AID_ROOT,      0, "system/bin/uncrypt" },
     { 00750, AID_ROOT,      AID_ROOT,      0, "system/bin/install-recovery.sh" },
-- 
1.9.1

